### PR TITLE
fix(cli): wing pack doesn't pack subdirectories

### LINKED ISSUE #4797: `wing pack` doesn't pack subdirectories
### I tried this:

See wing-containers - it has a bunch of files under subdirectories.

Trying to run `wing pack`.

I've tried to set `files` in package.json to no avail.

(tried `**/*.w`, list of directories, explicit list of files).

### This happened:

The resulting tarball doesn't contain the files from `sim/` or `tfaws/`.

### I expected this:

To work

### Is there a workaround?

_No response_

### Component

CLI

### Wing Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #4797

The root cause of this bug was that `wing pack` was incorrectly globbing files to copy into its staging directory. Instead of reinventing the wheel, I've changed the implementation to simply reuse the npm tooling that determine what files are included in a npm tarball, and use that to determine what should get copied into the staging directory.

I've also updated `wing pack` so it runs `wing compile` inside the staging directory instead of inside the user's original directory to reduce the chance for compile errors seeping into packaged Wing libraries.

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Review by MarkMcCulloh — APPROVED** [review]
🎉 

Added a label for e2e, mostly out of concern for the symlinking. Would be nice to run it through windows before merging.

File: apps/wing/src/commands/pack.ts [hunk 1]
```diff
@@ -176,7 +159,10 @@ async function withTempDir<T>(work: (workdir: string) => Promise<T>): Promise<T>
     return await work(workdir);
   } finally {
     process.chdir(cwd);
-    await fs.rm(workdir, { recursive: true });
+    // if you want to debug this you can keep the workdir around
+    if (!process.env.WING_PACK_KEEP_WORKDIR) {
```
Comment (eladb) [inline apps/wing/src/commands/pack.ts:163]: I think this should be the same as `--no-clean`

**Review by eladb — APPROVED** [review]
Ship it!

**Comment by eladb** [comment]
@Chriscbr resolved a couple of conversations to get this through for now. Please follow up.

File: apps/wing/src/commands/pack.ts [hunk 2]
```diff
@@ -133,34 +125,25 @@ export async function pack(options: PackageOptions = {}): Promise<string> {
...
-      await fs.copyFile(srcPath, destPath);
-    }
-  }
+/**
+ * Determine a list of files to copy to the staging directory, based on what npm says
+ * it would include in the tarball, given the user's package.json and a set of
+ * additional files that we want to include.
+ */
+async function listFilesToCopy(dir: string, extraGlobs: string[]): Promise<string[]> {
+  const arborist = new Arborist({ path: dir });
```
[the remark this answers is not in the document]
Comment (Chriscbr) [reply apps/wing/src/commands/pack.ts:134]: Good point, yeah I'm open to future optimizations here

File: apps/wing/src/commands/pack.ts (hunk 1 above)
Comment (Chriscbr) [reply apps/wing/src/commands/pack.ts:163]: got it, yeah I wasn't sure if it's something we wanted to add to our public API or not. Wanna propose a set of conventions for debug options / CLI commands in an issue?

**Pull request merged**